<a href="https://colab.research.google.com/github/AdhavanHero/Python-Machine-Learning/blob/main/KFold%20Cross%20Validation/KFold_Validation_on_Iris_flower_detection.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import numpy as np 
import pandas as pd
from sklearn.datasets import load_iris
iris = load_iris()
import math


In [ ]:
df = pd.DataFrame(iris.data, columns=iris.feature_names)
df.head()

sepal length (cm)  sepal width (cm)  petal length (cm)  petal width (cm)
0                5.1               3.5                1.4               0.2
1                4.9               3.0                1.4               0.2
2                4.7               3.2                1.3               0.2
3                4.6               3.1                1.5               0.2
4                5.0               3.6                1.4               0.2

In [ ]:
df['target'] = iris.target
df.head()
x = df.drop(['target'],axis='columns')
y=  df.target

In [ ]:
from sklearn.model_selection import train_test_split
x_train,x_test,y_train,y_test = train_test_split(x,y,test_size=0.2)

In [ ]:
from sklearn.ensemble import RandomForestClassifier
rf = RandomForestClassifier(n_estimators=46)
rf.fit(x_train,y_train)
rf.score(x_test,y_test)

0.8333333333333334

In [ ]:
 from sklearn.linear_model import LogisticRegression

lr = LogisticRegression()
lr.fit(x_train,y_train)
lr.score(x_test,y_test)

/usr/local/lib/python3.8/dist-packages/sklearn/linear_model/_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


0.8666666666666667

In [ ]:
 from sklearn.svm import SVC
svm = SVC()
svm.fit(x_train,y_train)
svm.score(x_test,y_test)

0.8666666666666667

In [ ]:
from sklearn import tree
dt =tree.DecisionTreeClassifier()
dt.fit(x_train,y_train)
dt.score(x_test,y_test)

0.8333333333333334

Logistic Regression


In [ ]:
#from this part is essential
from sklearn.model_selection import cross_val_score
l_scores = cross_val_score(LogisticRegression(), iris.data, iris.target,cv=3)
l_scores


/usr/local/lib/python3.8/dist-packages/sklearn/linear_model/_logistic.py:814: ConvergenceWarning: lbfgs failed to converge (status=1):
STOP: TOTAL NO. of ITERATIONS REACHED LIMIT.

Increase the number of iterations (max_iter) or scale the data as shown in:
    https://scikit-learn.org/stable/modules/preprocessing.html
Please also refer to the documentation for alternative solver options:
    https://scikit-learn.org/stable/modules/linear_model.html#logistic-regression
  n_iter_i = _check_optimize_result(


array([0.98, 0.96, 0.98])

In [ ]:
np.average(l_scores)

0.9733333333333333

Support Vectors

In [ ]:
b_scores = cross_val_score(SVC(), iris.data, iris.target,cv=3)
b_scores


array([0.96, 0.98, 0.94])

In [ ]:
np.average(b_scores)

0.96



Decision Tree


In [ ]:
c_scores = cross_val_score(tree.DecisionTreeClassifier(), iris.data, iris.target,cv=3)
c_scores

array([0.98, 0.94, 0.98])

In [ ]:
np.average(c_scores)

0.9533333333333334



Random Forest



In [ ]:
d_scores = cross_val_score(RandomForestClassifier(n_estimators=40), iris.data, iris.target,cv=3)
d_scores

array([0.98, 0.94, 0.94])

In [ ]:
np.average(d_scores)

0.9533333333333333

Best score so far is from Logistic Regression =0.9733333333333333


